### PR TITLE
envd: 0.3.0

### PR DESCRIPTION
**Pull request by frostming**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
  ==> /home/linuxbrew/.linuxbrew/Cellar/envd/0.2.6/bin/envd init
  There was an error generating build.envd: open /dev/tty: no such device or address
```

**Comment by frostming** [comment]
Due to behavior change and no other good replacement, I have to remove the test case for now(it still has a small test point).

**Comment by SMillerDev** [comment]
That's not a good enough test. We'll have to find a new one.

**Comment by frostming** [comment]
> That's not a good enough test. We'll have to find a new one.

That's fair, but other commands depend on docker, is that an acceptable testing infrastructure? Or we have to wait for the upstream until they provide a good one.

**Comment by SMillerDev** [comment]
Can we test a reliable error with the absence of docker?

File: Formula/envd.rb [hunk 1]
```diff
@@ -29,12 +29,16 @@ def install
...
+                                         shells: [:bash, :zsh])
   end
 
   test do
     output = shell_output("#{bin}/envd version --short")
     assert_equal "envd: v#{version}", output.strip
-    system bin/"envd", "init"
-    assert_predicate testpath/"build.envd", :exist?
+
+    _, stderr = Open3.capture3("#{bin}/envd env list")
```
Comment (SMillerDev) [inline Formula/envd.rb:41]:
```suggestion
    stderr = shell_output("#{bin}/envd env list 2>&1")
```

File: Formula/envd.rb [hunk 2]
```diff
@@ -29,12 +29,16 @@ def install
...
+                                         shells:                 [:bash, :zsh])
   end
 
   test do
     output = shell_output("#{bin}/envd version --short")
     assert_equal "envd: v#{version}", output.strip
-    system bin/"envd", "init"
-    assert_predicate testpath/"build.envd", :exist?
+
+    stderr = shell_output("#{bin}/envd env list 2>&1 || true")
```
Comment (SMillerDev) [inline Formula/envd.rb:41]: 15 can be whatever status code is expected.

```suggestion
    stderr = shell_output("#{bin}/envd env list 2>&1", 15)
```

Comment (frostming) [reply Formula/envd.rb:41]: why 15 not 1? Running in a clean docker env and it gives me `1`

Comment (SMillerDev) [reply Formula/envd.rb:41]: Because I had to guess something and wanted to make it clear that it was a number.

Comment (frostming) [reply Formula/envd.rb]: the ubuntu test image seems to have a different setup from others:
```
error: permission denied
```
Any suggestion on improving the test?

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**